### PR TITLE
Fix/company create offer

### LINKED ISSUE #144: Creating offer with company not working
After #138, companies cannot create offers since the middleware used only accounts for admin accounts.
https://github.com/NIAEFEUP/nijobs-be/blob/baef2d94f0beed9b59ba0d0e98eb5540f6055085/src/api/middleware/company.js#L38
In principle just replace `req.body.owner` with `req.user.company || req.body.owner` and add a test to verify this situation.

### PR DESCRIPTION
**Pull request by Naapperas**
Closes #144

### CODE REVIEW

**Comment by Naapperas** [comment]
Regarding the tests, I didn't add any specific tests for the added behavior, but rather updated one that circumvented the problem by setting manually the offer owner. With that change, it confirms the new middleware works. If I should add more tests or modify existing ones, please let me know.

File: src/api/middleware/auth.js [hunk 1]
```diff
@@ -76,6 +76,12 @@ const hasAdminPrivileges = async (req, res, next) => {
     return next();
 };
 
+const setOwnerCompany = (req, res, next) => {
+
+    req.ownerCompany = req.user?.company || req.body.owner;
```
[the remark this answers is not in the document]
Comment (imnotteixeira) [reply src/api/middleware/auth.js:81]: req.user?.company shouldn't be instead req.user?.company?._id.toString() ?

**Review by imnotteixeira — CHANGES_REQUESTED** [review]
As stated in #144, https://github.com/NIAEFEUP/nijobs-be/blob/1b277c87e024329ea5173b7b0e9fb9fd67fcb12a/src/services/offer.js#L231 might also need a change, by having it like so:

```js
if (offer?.isHidden && !(user?.isAdmin || offer.owner.toString() === req.targetOwner)) return null;
```

**Comment by Naapperas** [comment]
> As stated in #144,
> 
> https://github.com/NIAEFEUP/nijobs-be/blob/1b277c87e024329ea5173b7b0e9fb9fd67fcb12a/src/services/offer.js#L231
> might also need a change, by having it like so:
> 
> ```js
> if (offer?.isHidden && !(user?.isAdmin || offer.owner.toString() === req.targetOwner)) return null;
> ```

That line as already been changed by #154 to account for requests made by a god (that's what was pointed in #144).

https://github.com/NIAEFEUP/nijobs-be/blob/f710a3047f395b9790f9fe6424404cdc4122c148/src/services/offer.js#L237

Still, I'll apply this change to make it more coherent with the rest of the code.

File: src/api/middleware/auth.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Naapperas) [reply src/api/middleware/auth.js:81]: I thought so, but the issue stated that the replacement should be `req.user?.company || req.body.owner`, so I went with that.

[the remark this answers is not in the document]
Comment (Naapperas) [reply src/api/middleware/auth.js:81]: After looking through the code, i found this:

https://github.com/NIAEFEUP/nijobs-be/blob/f710a3047f395b9790f9fe6424404cdc4122c148/src/models/Account.js#L24-L25

So this should be kept as is.

[the remark this answers is not in the document]
Comment (imnotteixeira) [reply src/api/middleware/auth.js:81]: The thing is: the owner in the request will always be a string (which is an ObjectId, as long as it has the right format). Also, as you can see in https://github.com/NIAEFEUP/nijobs-be/blob/f710a3047f395b9790f9fe6424404cdc4122c148/src/services/offer.js#L237, the code always accesses the company id by accessing the user.company._id.toString(), therefore if we are to replace those occurrences with the targetOwner variable, it should keep the same logic I think, unless it doesn't work for some reason I am not understanding at the moment.

> the replacement should be req.user?.company || req.body.owner
this was a quick suggestion from me, take it with a grain of salt

**Comment by imnotteixeira** [comment]
> 
> 
> > As stated in #144,
> > https://github.com/NIAEFEUP/nijobs-be/blob/1b277c87e024329ea5173b7b0e9fb9fd67fcb12a/src/services/offer.js#L231
> > 
> > might also need a change, by having it like so:
> > ```js
> > if (offer?.isHidden && !(user?.isAdmin || offer.owner.toString() === req.targetOwner)) return null;
> > ```
> 
> That line as already been changed by #154 to account for requests made by a god (that's what was stated in #144).
> 
> https://github.com/NIAEFEUP/nijobs-be/blob/f710a3047f395b9790f9fe6424404cdc4122c148/src/services/offer.js#L237
> 
> Still, I'll apply this change to make it more coherent with the rest of the code.

Yes, the relevant part still requires updating, concerning the changes in this PR

**Review by imnotteixeira — APPROVED** [review]

**Pull request merged**